code/modules/reagents/chemistry/items.dm

```diff
@@ -162,3 +131,212 @@
+			var/obj/item/reagent_containers/container = I
+			if(!container.reagents.total_volume)
+				to_chat(user, "<span class='warning'>[container] is empty and can't be poured!</span>")
+				return
+
+			if(reagents.holder_full())
+				to_chat(user, "<span class='warning'>[src] is full.</span>")
+				return
+
+			var/trans = container.reagents.trans_to(src, container.amount_per_transfer_from_this, transfered_by = user)
+			to_chat(user, "<span class='notice'>You fill [src] with [trans] unit\s of the contents of [container].</span>")
+	if(I.heat < 1000)
+		return
+	set_lit(TRUE)
+	user.visible_message("<span class='notice'>[user] lights up the [src].</span>")
+
+/obj/item/burner/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(lit)
+		if(istype(target, /obj/item/reagent_containers))
+			var/obj/item/reagent_containers/container = target
+			container.reagents.expose_temperature(get_temperature())
+			to_chat(user, "<span class='notice'>You heat up the [src].</span>")
+			playsound(user.loc, 'sound/chemistry/heatdam.ogg', 50, TRUE)
+			return
+	else if(istype(target, /obj/item))
+		var/obj/item/item = target
+		if(item.heat > 1000)
+			set_lit(TRUE)
+			user.visible_message("<span class='notice'>[user] lights up the [src].</span>")
+
+/obj/item/burner/update_icon_state()
+	icon_state = "[initial(icon_state)][lit ? "-on" : ""]"
+
+
+/obj/item/burner/proc/set_lit(new_lit)
+	if(lit == new_lit)
+		return
+	lit = new_lit
+	if(lit)
+		force = 5
+		damtype = BURN
+		hitsound = 'sound/items/welder.ogg'
+		attack_verb_continuous = string_list(list("burns", "sings"))
+		attack_verb_simple = string_list(list("burn", "sing"))
+		START_PROCESSING(SSobj, src)
+	else
+		hitsound = "swing_hit"
+		force = 0
+		attack_verb_continuous = null //human_defense.dm takes care of it
+		attack_verb_simple = null
+		STOP_PROCESSING(SSobj, src)
+	set_light_on(lit)
+	update_icon()
+
+/obj/item/burner/extinguish()
+	set_lit(FALSE)
+
+/obj/item/burner/attack_self(mob/living/user)
+	if(user.is_holding(src))
+		if(lit)
+			set_lit(FALSE)
+			user.visible_message("<span class='notice'>[user] snuffs out [src]'s flame.</span>")
+	else
+		. = ..()
+
+/obj/item/burner/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(lit && M.IgniteMob())
+		message_admins("[ADMIN_LOOKUPFLW(user)] set [key_name_admin(M)] on fire with [src] at [AREACOORD(user)]")
+		log_game("[key_name(user)] set [key_name(M)] on fire with [src] at [AREACOORD(user)]")
+	return ..()
+
+/obj/item/burner/process()
+	var/current_heat = 0
+	var/number_of_burning_reagents = 0
+	for(var/_reagent in reagents.reagent_list)
+		var/datum/reagent/reagent = _reagent
+		if(istype(reagent, /datum/reagent/consumable/ethanol))
+			current_heat += 2193//ethanol burns at 1970C (at it's peak)
+			number_of_burning_reagents += 1
+			reagents.remove_reagent(/datum/reagent/consumable/ethanol, 0.05)
+			continue
+
+		if(ispath(reagent, /datum/reagent/fuel))
```

Comment:
I see mixed `istype()` and `ispath()`
Sure this is correct?

Also, could these istype checks be moved to burning properties, or would that be too complex / unwarranted?

Reply:
What do you mean by burning properties? You mean as part of the reagent datum? I suppose making a flammable component to reagents would be a neat idea thinking about it, but that's probably out of scope for this pr!

Oh, and the ispath is used to ensure that no children pass, as oil is a child of fuel.

Reply:
Done